bluemira/equilibria/coils/_grouping.py

```diff
@@ -460,7 +494,7 @@ def get_max_current(self, max_current: float = np.infty) -> np.ndarray:
         np.ndarray
         """
         return np.where(
-            np.isnan(self.j_max),
+            np.isnan(self.j_max) ^ ~self._flag_sizefix,  # or not
```

Comment:
Wouldn't 'or not' be ` x | ~y`, not `x ^ ~y`?

Also, isn't `x ^ ~y` equivalent to `x == y`? Which I think could make the intent clearer

Reply:
Your right it should be `x | ~y` 

on your second point no not quite because for a XOR -> NOT (XNOR) gate `False + False = True`

Reply:
Not sure what you mean on the second point, both `x ^ ~y` and `x == y` return `True` for `x=False, y=False`.

```pycon
In [1]: import numpy as np

In [2]: x = np.array([True, True, False, False])

In [3]: y = np.array([True, False, True, False])

In [4]: x ^ ~y
Out[4]: array([ True, False, False,  True])

In [5]: x == y
Out[5]: array([ True, False, False,  True])
```

Or am I misunderstanding something?

Reply:
Just because I kept second guessing myself

j_max is nan -> use max current
not size fixed -> use max current

|RAW|||WITH NOT CONVERSION| ||| 
|--|--|--|--|--|--|--| 
|F   |T | F|    -> |  F | F | F|
|T  |F|   T   |-> |  T  |T |  T|
|T | T  | T  | ->  | T | F  | T|
|F|  F  | T |  ->   |F | T |  T|

Reply:
Ahh not proper AND apparently. In that case yes they are the same

https://en.wikipedia.org/wiki/And_gate